### DISCUSSION TITLE
Faculdades, certificados e cursos de aprendizado

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#39] [Discussão]
Galera os cursos do Coursera vale a pena ? Ou garimpar online ? Ter certificados em Universidades Americanas, Asiaticas e Europeias vale a pena ou não ? O que importa e ter o conhecimento mesmo e desenvolver projetos pessoais ou comerciais mesmo ?? Na hora de uma entrevista qual a diferença e impacto no entrevistador de ter projetos e tambem certificados ?

exs:

https://www.coursera.org/specializations/full-stack
https://www.coursera.org/learn/angular-js
https://www.coursera.org/learn/meteor-development
https://www.coursera.org/specializations/web-app-architecture

O coursera toda semana me manda cursos que me interessa seja profissionalmente ou por hobby ( musica, design , 3d etc) 

E tudo que eu sei hoje eu aprendi meio que na raça.. Vale a pena investir em Coursera e outros cursos pagos de fora, ou investir em cursos daqui, ou ir na raça mesmo e desenvolver projetos comerciais e pessoais dá na mesma no final ??

Na minha visão pessoal, fazer um curso e nao ter prosseguimento depois é perda de tempo e dinheiro. Eu fiz cursos de C# , que não tive após demanda de back, e foi bom aprender pra mexer e ter noções mas eu senti que não fui na profundidade que eu queria.

Hoje eu trabalho numa agência e não tenho abertura pra aprender e desenvolver ferramentas ,linguagens e metodologias que eu gostaria de aprender como React,Angular,NodeJS,Ruby,Mongo etc etc, todas eu ja vi um pouco por "curiosidade", mas nao tenho a profundidade que eu gostaria de ter e nem o dominio.

Num escopo de curto-médio prazo (2 anos) eu queria estar nesse ritmo pelo menos ao que toca a projeto pessoais, mesmo sobrando muito pouco tempo por semana (menos de 6h) , pra aprender isso tudo.

O ideal seria arranjar um espaço que eu conseguisse aprender e trabalhar tambem. Mas empregos assim são raros. 

Enfim é um planejamento meu, porque as vagas que eu miro em Linkedin,StackOverflow etc tudo pedem o dominio de coisas que eu "só vi por cima" , e por isso não me sinto seguro de ir atras.. mas esse ano quero me organizar para ter condições de, e estou criando essa issue, pq eu aposto que tem muitos desenvolvedores como eu, que tem pouco tempo pra desenvolver e investir na carreira.

Pelo que eu vejo a maioria das empresas esta focada em finalizar projetos e investir pouco nos funcionarios ou nada. Na verdade em quanto PJ, o conhecimento é por parte do colaborador, se ele trabalha 9-10-12 horas por dia e quase nao sobra tempo pra estudo, especialmente se tem filho,familia etc, fica dificil.

Eu to nesse ritmo a alguns anos e quero mudar isso.. (desculpem o desabafo) , mas é pra ter uma direção e opinião externa alem da minha intuição.

Valeu.

**Comment by rubenmarcus** [comment]
Outro desabafo, que eu estava falando com meu pai esses dias, nao importa se a cada ano eu aumento meu salario em 500 ou 1000 reais, ou se daqui 2 anos eu vou estar ganhando 6,8,10mil, mas se o que eu faço e praticamente a mesma coisa a anos. Recortar layouts e implementar javascripts e funcionalidades. Nao agrega conhecimento e nada pessoal. Poder trabalhar com marcas e empresas grandes, ter um salario razoavel, mas pessoalmente e intelectualmante o seu potencial estar sendo subaproveitado.. Ver que quase uma decada inteira ( eu trabalho desde os 21 tenho 26) , voce simplesmente fez as mesmas coisas. 

Ou como meu pai diz "viver pra apagar fogo" 

Qual a importância de cursos e projetos pessoais nesse contexto todo ??

Que em questão de potencial de aprendizado e implementação eu sinto que eu consigo desenvolver desde projetos gráficos ( Web,Artes,3d,Id Visual,Tipografia) , até programação front e back.. e sinto que estou perdendo tempo recortando layouts...

**Comment by designernatan** [comment]
Acredito que tudo depende do seu objetivo.

Sei que para muitas consultorias, alguns certificados são imprescindíveis, uma vez que o cliente pede que elas tenham X funcionários com a certificação Y.

Exceto esse nicho (posso chamar de nicho?) acho que nem vale tanto assim correr atrás de certificados.

**Comment by felipefialho** [comment]
É uma boa pergunta. Até hoje não vi nenhuma empresa exigindo qualquer tipo de certificado parte Desenvolvedores Front-End.

Até mesmo faculdade é um tema controverso.

Como disse o @ntnleao tudo depende dos seus objetivos, mas acredito que essa necessidade de certificados e etc, sejam mais comum em multinacionais muito específicas.

**Comment by fernandofleury** [comment]
Nao precisa. Nao tem muito bem como resumir isso, mas nao precisa. No máximo o que voce vai esbarrar é a necessidade de um diploma.

Eu vou ser polemico aqui, e se voce acabar caindo em um lugar que considera isso necessário, provavelmente nao é um lugar que compreende o que é o seu cargo.

**Comment by doug2k1** [comment]
Também não vejo que os certificados em si sejam um diferencial, mas estudar coisas por fora é sempre bom. Se você aprende algo novo e faz um projeto pessoal com aquilo pra exercitar e bota no GitHub, ou faz um post em seu blog sobre aquilo, acaba contribuindo pra incrementar seu portfólio, e isso sim gera visibilidade e é bem visto por quem vai contratar. Além de expandir seus horizontes e te ajudar no seu trabalho atual mesmo.

Eu sempre estou fuçando sites de cursos como Coursera, Udacity, Udemy, Edx, Lynda, Pluralsight, Code School e outros do tipo, procurando cursos interessantes pra fazer.

**Comment by rubenmarcus** [comment]
Legal @doug2k1 voce paga os 60 dolares para lo certificado?  Vejo uma galera no linkedin que poem que fez cursos alura, caellum etc..  mesmo se tratando de curso livre e  essas escolas ser boas 60 dólares nao sao 2500 reais(curso Javascript e Angular Caellum)  e o um certificado da Universidade do Novo mexico,  da Universidade de Hong Kong,  Tokyo etc deve valer  alguma coisa..  Eu sei que nao e o preponderante ter a experiência  e conhecimento e muito mais mas e so um ponto que estou levantando

**Comment by doug2k1** [comment]
Nunca peguei um certificado desses cursos. De repente, esses cursos de faculdade, onde vc é mais exigido, tem que entregar trabalhos, etc. realmente sejam mais reconhecidos.

**Comment by tcelestino** [comment]
Não sei se uma faculdade vai dizer muita coisa, apesar que quando fiz aprendi bastante, principalmente a parte teórica do desenvolvimento em geral. Certificados? Existem pessoas que possuem diversos certificados, cursos etc... e não é tão bom, logo, acredito que o que mais vale é mostrar que você realmente sabe.

@rubenmarcus fiquei meio que intrigado com isso: 

"...Hoje eu trabalho numa agência e não tenho abertura pra aprender e desenvolver ferramentas ,linguagens e metodologias que eu gostaria de aprender como React,Angular,NodeJS,Ruby,Mongo etc etc..."

Não pense apenas em aplicar isso no trabalho, pense em algo e comece a estudar a partir dessa ideia. Também tinha esse pensamento de apenas aprender para aplicar nas empresas que trabalhei, depois vi que muitas não vão valorizar tanto assim e é mais enriquecedor você aprender além de apenas aplicar em projetos no qual é remunerado.

**Comment by felipefialho** [comment]
Perfeito @fernandofleury. Tenho a mesma opinião.

> Eu vou ser polemico aqui, e se voce acabar caindo em um lugar que considera isso necessário, provavelmente nao é um lugar que compreende o que é o seu cargo.

**Comment by bernardodiasc** [comment]
Faz um projeto que explora as tenicas mais interessantes que voce conhece e põe no GitHub, vale muito mais que qualquer certificado. Ou artigos, ou qualquer coisa que vc possa usar de "show off" a respito das suas habilidades.

No mercado de trabalho, na pratica, estamos falando de pessoas, e não tem como uma pessoa ler a mente da outra e de cara conhecer todas as potencias da outra, as vezes a opotunidade de mostrar seu valor não acontece, as vezes mostrar seu valor não é tão simples, requer muita conversa e muita ação e talvez só quem te conhece de longa data reconhece isso em voce, por exemplo. Os certificados e comprovações de que voce realmente estudou determinado assunto e (talvez) sabe sobre aquilo é uma forma de facilitar a oportunidade de alguem que precisa daquela habilidade. Mas não é a única forma de mostrar seu valor.

Acho que em muitos casos onde o resultado do nosso trabalho não afeta a integridade das pessoas, as cobranças e exigencias são mais suaves. Já vi programador trabalhando em sistemas de telefonia com multas milionárias em caso de downtime, se eu fosse dono de um sistema assim, com certeza iria contratar pelos certificados mais relevantes possíveis. Digo, alguns casos a seriedade do projeto é tão grande, que entra um outro tipo de profissional.

**Comment by vitorbritto** [comment]
Sou da seguinte opinião:
- Faculdade é importante para entender a base da Ciência da Computação / Engenharia da Computação / Sistema da Informação. Mas isso não quer dizer que você não possa aprender sozinho. O importante é ler e praticar.
- Um diploma te serve como uma carta na manga para um salário mais avantajado e determinados direitos civis. Quem tem Ensino Superior, possui mais "garantias" no Brasil. Acho ridículo essa separação de _status_, mas...
- Prefiro contratar/trabalhar com/terceirizar um dev com experiência e sem diploma, do que contratar/trabalhar com/terceirizar um "diplomado" sem experiência. Para mim, o salário vai de acordo com o que o cara for capaz de fazer/exercer. E, o mais importante, se for um profissional esforçado, eu faço questão de ensinar o que o cara não souber. E se eu não souber, faço questão (mais ainda) de aprender para ensinar ao cara. 
- Não pense no certificado, pense em como você vai aprimorar seu conhecimento nos cursos extras. Se uma empresa te exige certificados para provar que você conhece uma tecnologia/ferramenta/linguagem A,B ou C, o que posso dizer? "Talk is cheap. Show me the code".
- Não se torne refém do tempo, nem do seu local de trabalho. Faça experimentos nos horários vagos e construa algo que você e as pessoas tenham necessidade em utilizar. O primeiro passo já foi dado, você está no GitHub, interagindo com outros desenvolvedores. Coloca a mão na massa e criar algo para você a que ajude as pessoas a sua volta também.
  Sou da seguinte opinião:
- Faculdade é importante para entender a base da Ciência da Computação / Engenharia da Computação / Sistema da Informação. Mas isso não quer dizer que você não possa aprender sozinho. O importante é ler e praticar.
- Um diploma te serve como uma carta na manga para um salário mais avantajado e determinados direitos civis. Quem tem Ensino Superior, possui mais "garantias" no Brasil. Acho ridículo essa separação de _status_, mas...
- Prefiro contratar/trabalhar com/terceirizar um dev com experiência e sem diploma, do que contratar/trabalhar com/terceirizar um "diplomado" sem experiência. Para mim, o salário vai de acordo com o que o cara for capaz de fazer/exercer. E, o mais importante, se for um profissional esforçado, eu faço questão de ensinar o que o cara não souber. E se eu não souber, faço questão (mais ainda) de aprender para ensinar ao cara. 
- Não pense no certificado, pense em como você vai aprimorar seu conhecimento nos cursos extras. Se uma empresa te exige certificados para provar que você conhece uma tecnologia/ferramenta/linguagem A,B ou C, o que posso dizer? "Talk is cheap. Show me the code".
- Não se torne refém do tempo, nem do seu local de trabalho. [Faça experimentos nos horários vagos](https://github.com/vitorbritto/labs) e [construa algo que você e as pessoas tenham necessidade em utilizar](https://github.com/vitorbritto/just). O primeiro passo já foi dado... você está no GitHub interagindo com outros desenvolvedores. =]

**Comment by suissa** [comment]
Cara se quiser certificado o Be MEAN dpá e de curso completo pra um especialista em desenvolvimento web é o único curso gratuito e com o melhor conteúdo que você poderia encontrar por aí.

youtube.com/c/webschool-io

**Comment by rubenmarcus** [comment]
Bem legal as dicas pessoal valeu mesmo @suissa @LFeh @vitorbritto  @ntnleao  @bernardodiasc  @tcelestino , Uma coisa que eu tenho muita vontade em fazer é ensinar mulecada pobre a desenvolver. Nem que o básico de design e desenvolvimento. Até falei com a Vanessa que palestrou no FrontinSampa q ela fazia trampo voluntario em uma casa espirita.. queria poder passar um pouco do conhecimento a quem não tem acesso de forma alguma... Eu moro na periferia de SP e vejo a mulecada sem perspectiva nenhuma pensando em virar cantor de funk ou qualquer outra coisa e acho foda, mesmo que nenhum virasse dev, mas ja apontar a direção e dar um conhecimento minimo ja seria legal.

Numa startup que trabalhei um dos analistas dava aula na FITO em Osasco e falou de eu dar aula pro ensino médio tecnico mas não rolou.

**Comment by dionarodrigues** [comment]
Ótima explicação @vitorbritto . 

Me lembro que quando cheguei em sampa, para morar, há 3 anos, me rejeitaram em uma entrevista, em um banco, por não ter diploma. Foi a única vez que o diploma fez falta em minha carreira profissional. 

Por isso, para evitar um futuro problema, estou fazendo um curso tecnológico; algo rápido, só pelo diploma. 

Claro que, como em qualquer curso, sempre tem algo para eu aprender. Da mesma forma, se eu quiser me aprofundar em algum tema novo, terei que me esforçar sozinho, fora do ambiente em aula, pois faculdade só ensina o básico (só mostra o caminho), e muitas vezes distoa bastante do mercado de trabalho, da realidade. 

Mas, ao mesmo tempo, penso que é uma ótima porta de entrada para aqueles que não tem muita noção de alguma coisa ou que querem uma oportunidade no mercado de trabalho: estágios pode ser um trampolim.

---

Recentemente o IBGE lançou um concurso para front-ends com um salário de 7k. Para isso, também é necessário diploma. Portanto, sou dos que pensam q o diploma não o faz um profissional melhor do que outro que não o tem, mas pode ser um diferencial. ;)

**Comment by felipefialho** [comment]
Vou copiar o que escrevi no fórum do Tableless, pode enriquecer a discussão

---

Minha opinião... e tão somente minha opinião.

Fiz uma faculdade tecnóloga de 2 anos na UNIP, curso fraco, mas acreditem se quiser, me da direito a fazer pós graduação. Terminei essa faculdade com apenas 18 anos (comecei com 16 pois terminei cedo o ensino médio).

2 anos após terminar essa faculdade comecei uma de Sistemas de Informação, na qual cursei 1 ano e meio (terminaria com 24), naquela época achava que seria sensacional ter 2 faculdades aos 24 anos. Porém acabei não conseguindo levar tão a sério (chegava muito cansado) e tranquei.

Eu terminaria esse curso no meio do ano passado, e nesse período (entre 2011 e 2014), só para citar algumas coisas:

2012 - Criei um [Eric Cartman em CSS](http://codepen.io/LFeh/pen/qzDCJ)
2013 - Criei o [meu site](http://www.felipefialho.com/)
2013 - Criei um [Piano usando CSS e JavaScript](http://www.felipefialho.com/piano/)
2013 - Criei um [Xadrez usando CSS e JavaScript](http://chessjs.trendi.com.br/)
2014 - Criei o [CSS Components](http://www.felipefialho.com/css-components/)
2014 - Criei o Meetup CSS junto com o @raphaelfabeni

E estudei, estudei e estudei muito.

No meio de 2014, quando estaria terminando a faculdade, comecei relações com uma Agencia de NYC, onde estou trabalhando remoto hoje.

Sabe o que quero dizer? Que duvido muito que conseguisse fazer metade dessas coisas se tivesse continuado a faculdade, e isso sem dúvidas impactaria no meu nível profissional atual. Não acredito que eu teria aprendido na faculdade nem perto do que aprendi estudando e criando projetos.

Apesar de eu ter feito essa faculdade tecnóloga, nunca peguei o diploma (a faculdade liga cobrando as vezes), nenhuma empresa até hoje fez essa exigência, se um dia precisar eu pego.

Ou seja, acredito ser muito relativo. A faculdade me parece mais útil no inicio de carreira, quando você precisa de um estágio e é difícil conseguir uma vaga júnior sem experiencia.

Também acho que seja útil para quem não consegue focar nos estudos sozinho, nesse caso faculdade ou cursos técnicos (olha que prefiro essa opção) também são boas pedidas.

**Comment by tcelestino** [comment]
@vitorbritto :+1:

**Comment by yuricamara** [comment]
Necessário não é mas pode fazer diferença.  Eu lembro que eu consegui meu primeiro emprego de front exaltando os cursos que havia feito. Tudo faz diferença. É uma competição.  Só que tem coisas que tem peso grande e outras coisas pequeno.

**Comment by nighto** [comment]
Até hoje nunca foi exigido nas empresas onde trabalhei. Porém entendo a sua importância, especialmente para trabalhar no exterior (alguns países exigem para emitir o visto).

**Comment by dionarodrigues** [comment]
Aproveitando essa issue sobre cursos, vocês conhecem a didática e a intensidade do curso de JS das escolas [Caelum ](https://www.caelum.com.br/curso-javascript-jquery/)e [DRC](http://drc.com.br/cursos/javascript-com-ajax)? Indicam para um nível intermediário ou limitam-se apenas ao básico?

**Comment by sergiokopplin** [comment]
Centralizando meu comentário da outra issue.

---

@felipesousa bom dia!
Cara, eu terminei a minha na FMU em SP. 
Fiz Sistemas para Internet, porém quando comecei eu já trabalhava com desenvolvimento, logo a faculdade era mais pra "ter" a graduação.

Eu não aprendi muito sobre o desenvolvimento em sí, como programar e tal. Mas aprendi muito sobre projetos, arquitetura e redes.

A faculdade não foi muito boa, o curso pecou em alguns momentos, mas eu acredito que faria novamente por ter conhecido tanta gente boa.

Acredito sim que vale a pena e que precisamos sair um pouco do nosso mundo da internet e ver pessoas de verdade!

---

**Vantagens**: o curso me proporcionou conhecimentos diversos que irão me ajudar no mundo do front-end.
**Desvantagens**: não existe uma matéria voltada a isso e nem que estude as tecnologias mais recentes, logo é uma grande desvantagem!

---

Resumo: _vale a pena_. Mas precisa ser analisado muito bem antes de iniciar os estudos.
**Avaliar** a faculdade, grade, professores e infra da faculdade.

**Comment by Fabio-Pina** [comment]
> Vou copiar o que escrevi no fórum do Tableless, pode enriquecer a discussão
> 
> Minha opinião... e tão somente minha opinião.
> 
> Fiz uma faculdade tecnóloga de 2 anos na UNIP, curso fraco, mas acreditem se quiser, me da direito a fazer pós graduação. Terminei essa faculdade com apenas 18 anos (comecei com 16 pois terminei cedo o ensino médio).
> 
> 2 anos após terminar essa faculdade comecei uma de Sistemas de Informação, na qual cursei 1 ano e meio (terminaria com 24), naquela época achava que seria sensacional ter 2 faculdades aos 24 anos. Porém acabei não conseguindo levar tão a sério (chegava muito cansado) e tranquei.
> 
> Eu terminaria esse curso no meio do ano passado, e nesse período (entre 2011 e 2014), só para citar algumas coisas:
> 
> 2012 - Criei um [Eric Cartman em CSS](http://codepen.io/LFeh/pen/qzDCJ)
> 2013 - Criei o [meu site](http://www.felipefialho.com/)
> 2013 - Criei um [Piano usando CSS e JavaScript](http://www.felipefialho.com/piano/)
> 2013 - Criei um [Xadrez usando CSS e JavaScript](http://chessjs.trendi.com.br/)
> 2014 - Criei o [CSS Components](http://www.felipefialho.com/css-components/)
> 2014 - Criei o Meetup CSS junto com o @raphaelfabeni
> 
> E estudei, estudei e estudei muito.
> 
> No meio de 2014, quando estaria terminando a faculdade, comecei relações com uma Agencia de NYC, onde estou trabalhando remoto hoje.
> 
> Sabe o que quero dizer? Que duvido muito que conseguisse fazer metade dessas coisas se tivesse continuado a faculdade, e isso sem dúvidas impactaria no meu nível profissional atual. Não acredito que eu teria aprendido na faculdade nem perto do que aprendi estudando e criando projetos.
> 
> Apesar de eu ter feito essa faculdade tecnóloga, nunca peguei o diploma (a faculdade liga cobrando as vezes), nenhuma empresa até hoje fez essa exigência, se um dia precisar eu pego.
> 
> Ou seja, acredito ser muito relativo. A faculdade me parece mais útil no inicio de carreira, quando você precisa de um estágio e é difícil conseguir uma vaga júnior sem experiencia.
> 
> Também acho que seja útil para quem não consegue focar nos estudos sozinho, nesse caso faculdade ou cursos técnicos (olha que prefiro essa opção) também são boas pedidas.

Eu gostaria de ajudar ou até mesmo pegar projetos para desenvolver. Vi até alguns mas não sei como iniciá-los, tem alguma dica com relação a isso?
